core/store/src/trie/mod.rs

```diff
@@ -540,6 +549,21 @@ impl WrappedTrieChanges {
     ) -> Result<(), Box<dyn std::error::Error>> {
         self.trie_changes.deletions_into(self.trie.clone(), store_update)
     }
+
+    pub fn key_value_changes_into(
+        &self,
+        store_update: &mut StoreUpdate,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        store_update.trie = Some(self.trie.clone());
+        for (key, changes) in &self.kv_changes {
+            let mut storage_key = Vec::with_capacity(self.block_hash.as_ref().len() + key.len());
```

Comment:
We should probably rename `StateChangeCause::NonFinalizable` to `StateChangeCause::NotWritableToDisk` and add an assertion here that makes sure it is not in`changes`.